services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceControllerClusterConfig.java

```diff
@@ -894,7 +904,13 @@ public VeniceControllerClusterConfig(VeniceProperties props) {
     this.emergencySourceRegion = props.getString(EMERGENCY_SOURCE_REGION, "");
     this.allowClusterWipe = props.getBoolean(ALLOW_CLUSTER_WIPE, false);
     this.concurrentInitRoutinesEnabled = props.getBoolean(CONCURRENT_INIT_ROUTINES_ENABLED, false);
-    this.controllerInAzureFabric = props.getBoolean(CONTROLLER_IN_AZURE_FABRIC, false);
+    this.controllerClusterHelixCloudEnabled = props.getBoolean(CONTROLLER_CLUSTER_HELIX_CLOUD_ENABLED, false);
+    this.controllerStorageClusterHelixCloudEnabled =
+        props.getBoolean(CONTROLLER_STORAGE_CLUSTER_HELIX_CLOUD_ENABLED, false);
+    this.controllerHelixCloudProvider = props.getString(CONTROLLER_HELIX_CLOUD_PROVIDER, "");
+    this.controllerHelixCloudId = props.getString(CONTROLLER_HELIX_CLOUD_ID, "");
+    this.controllerHelixCloudInfoProcessorName = props.getString(CONTROLLER_HELIX_CLOUD_INFO_PROCESSOR_NAME, "");
+    this.controllerHelixCloudInfoSources = props.getList(CONTROLLER_HELIX_CLOUD_INFO_SOURCES, Collections.emptyList());
```

Comment:
We need to handle the empty string issue here as well, right?

Reply:
Yes. Good catch.